plugins/modules/files/sapcar_extract.py

```diff
@@ -0,0 +1,220 @@
+        If this parameter is not provided the module will look in C(PATH).
+    type: path
+  signature:
+    description:
+      - If C(true) the signature will be extracted.
+    default: false
+    type: bool
+  manifest:
+    description:
+      - The name of the manifest.
+    default: "SIGNATURE.SMF"
+    type: str
+  remove:
+    description:
+      - If C(true) the SAR/CAR file will be removed. B(This should be used with caution!)
+    default: false
+    type: bool
+author:
+    - Rainer Leber (@RainerLeber)
+notes:
+    - Always returns B(changed True) in C(check_mode).
+'''
+
+EXAMPLES = """
+- name: Extract SAR file
+  community.general.sapcar_extract:
+    path: "~/source/hana.sar"
+
+- name: Extract SAR file with destination
+  community.general.sapcar_extract:
+    path: "~/source/hana.sar"
+    dest: "~/test/"
+
+- name: Extract SAR file with destination and download from webserver can be a fileshare as well
+  community.general.sapcar_extract:
+    path: "~/source/hana.sar"
+    dest: "~/dest/"
+    binary_path: "https://myserver/SAPCAR"
+
+- name: Extract SAR file and delete SAR after extract
+  community.general.sapcar_extract:
+    path: "~/source/hana.sar"
+    remove: true
+
+- name: Extract SAR file with manifest
+  community.general.sapcar_extract:
+    path: "~/source/hana.sar"
+    signature: true
+
+- name: Extract SAR file with manifest and rename it
+  community.general.sapcar_extract:
+    path: "~/source/hana.sar"
+    manifest: "MyNewSignature.SMF"
+    signature: true
+"""
+
+import os
+from tempfile import NamedTemporaryFile
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url
+from ansible.module_utils._text import to_native
+
+
+def get_list_of_files(dir_name):
+    # create a list of file and directories
+    # names in the given directory
+    listOfFile = os.listdir(dir_name)
+    allFiles = list()
+    # Iterate over all the entries
+    for entry in listOfFile:
+        # Create full path
+        fullPath = os.path.join(dir_name, entry)
+        # If entry is a directory then get the list of files in this directory
+        if os.path.isdir(fullPath):
+            allFiles = allFiles + [fullPath]
+            allFiles = allFiles + get_list_of_files(fullPath)
+        else:
+            allFiles.append(fullPath)
+    return allFiles
+
+
+def download_SAPCAR(binary_path):
+    bin_path = None
+    is_random = False
+    random_file = NamedTemporaryFile(delete=False)
+    # download sapcar binary if url is provided otherwise path is returned
+    if binary_path is not None:
+        if binary_path.startswith('http://' or 'https://'):
+            with open_url(binary_path) as response:
+                with open(random_file, 'wb') as out_file:
+                    data = response.read()
+                    out_file.write(data)
+            os.chmod(random_file, 0o700)
+            bin_path = random_file
+            is_random = True
```

Comment:
Instead of returning this flag, use:
```suggestion
            module.add_cleanup_file(bin_path)
```
to make `AnsibleModule` clean up the file on exit.